src/Clash/Cores/Ethernet/Examples/FullUdpStack.hs

```diff
@@ -0,0 +1,282 @@
+   We use the function `routeBy`, which makes a common design pattern much simpler to implement, namely we pattern match on the value of one of the fields of the `_meta` field: ARP packets have `_etherType` 0x0806, `_etherType` 0x0800. 
+   See the type of `routeBy` for the exact type. This splits the stream up into a vector of two signals: `[arpEthIn, ipEthIn]`.
+2. `ipEthIn` is then passed to `ipDepacketizerLiteC`, which is a convenience composition of the `ipDepacketizer` with `toLiteC`: it takes a `PacketStream dom n EthernetHeader` and outputs a `PacketStream dom n IPv4HeaderLite` by stripping off the EthernetHeader (`ipDepacketizer`) and then extracting the `Ipv4Header` in the metadata to an `IPv4HeaderLite` (`toLiteC`).
+   Using a locally define `isForMyIp`, we filter out all packets that do not have either the subnet's broadcast IPv4 address or my IPv4 address as `_ipv4lDestination` field in the `_meta` field.
+   This is then passed to the `icmpUdpStack`, which is preceded and followed by a packet buffer (`packetBufferC`). a circuit which stores words in a buffer until the packet is complete once a packet is complete it will send the entire packet out at once without stalls. If a word in a packet has `_abort` set to true, the packetBuffer will drop the entire packet.  
+   These packet buffers have a depth of 2^10 data bits and a depth of 2^4 meta bits. Since one packet may consist of many fragments, which by definition must have the same metadata, but have different content, we buffer metadata separately from content.
+   The reason for using packet buffers is because they absorb some backpressure generated by the @ipTx port  and the icmpUdpStack. In particular, the `toEthernetStreamC` asserts backpressure until arpC has given an `ArpResponse` to its `ArpLookup`.
+   From the packetbuffer, the stream is passed to `ipLitePacketizerC`, which is another convenience composition of `fromLiteC` and `ipPacketizerC`. `fromLiteC` simply maps the `_meta` field of the packet stream from an `IPv4HeaderLite` to an `IPv4Header`, while `ipPacketizer` packetizes the `IPv4Header` into the data of the packet stream, and puts the destination IP address into the `_meta` field. 
+   This is convenient since the following ARP lookup service can take the required IP address directly from the `_meta`.
+3. The `toEthernetStreamC` performs the ARP lookup and, if it gets an `ArpEntryFound` from `arpC` (in the backward line of the arplookup port), it constructs an `EthernetHeader` with the source and target MAC fields set to our MAC address and the found MAC address respectively, and the appropriate ethertype for IPv4.
+4. Finally, a `packetArbiterC` running in `RoundRobin` mode switches in a round-robin fashion between outputting a packet from the @arpEthOut stream and the @ipEthOut stream, i.e.
+   it alternatingly will output a complete packet coming from `toEthernetC` and a complete packet from `icmpUdpPacket`.
```

Comment:
> it will alternatingly will output a complete packet
Well, not completely, because the packet arbiter forwards data from the same input while it has data, iirc. The collect mode just determines how the next source is picked once the selected source has run out of packets.